cg/meta/orders/ticket_handler.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import re
+from typing import Optional
+
+from cg.apps.osticket import OsTicket
+from cg.exc import TicketCreationError
+from cg.server.schemas.order import OrderIn
+from cg.store import Store, models
+
+LOG = logging.getLogger(__name__)
+
+
+class TicketHandler:
+    """Handle tickets in the meta orders context"""
+
+    NEW_LINE = "<br />"
+
+    def __init__(self, osticket_api: OsTicket, status_db: Store):
+        self.osticket: OsTicket = osticket_api
+        self.status_db: Store = status_db
+
+    @staticmethod
+    def parse_ticket_number(name: str) -> Optional[int]:
+        """Try to parse a ticket number from a string"""
+        # detect manual ticket assignment
+        ticket_match = re.fullmatch(r"#([0-9]{6})", name)
+        if ticket_match:
+            ticket_number = int(ticket_match.group(1))
+            LOG.info("%s: detected ticket in order name", ticket_number)
+            return ticket_number
+        LOG.info("Could not detected ticket number in name %s", name)
+        return None
+
+    def create_ticket(
+        self, order: OrderIn, user_name: str, user_mail: str, project: str
+    ) -> Optional[int]:
+        """Create a ticket and return the ticket number"""
+        message = self.create_new_ticket_message(order=order, user_name=user_name, project=project)
+        ticket_nr = None
+        try:
+            ticket_nr: Optional[int] = self.osticket.open_ticket(
+                name=user_name,
+                email=user_mail,
+                subject=order.name,
+                message=message,
+            )
+
+            LOG.info(f"{ticket_nr}: opened new ticket")
+        except TicketCreationError as error:
+            LOG.warning(error.message)
+
+        return ticket_nr
+
+    def create_new_ticket_message(self, order: OrderIn, user_name: str, project: str) -> str:
+        message = f"data:text/html;charset=utf-8,New incoming {project} samples: "
+
+        for sample in order.samples:
```

Comment:
is OrderSample going to be a class? Or in next PR? Its also an ambiguous dict right now

Reply:
Yes next pr